src/compartment.c

```diff
@@ -548,10 +329,13 @@ comp_map(struct Compartment *to_map)
         PROT_READ | PROT_WRITE | PROT_EXEC, // TODO fix this
         MAP_PRIVATE | MAP_FIXED | MAP_ANONYMOUS | MAP_STACK, -1, 0);
     to_map->stack_pointer = to_map->scratch_mem_stack_top;
-    assert(map_result != MAP_FAILED);
+    if (map_result == MAP_FAILED)
+    {
+        err(1, "Error mapping compartment %zu stack!\n", to_map->id);
+    }
 
     // Inject intercept instructions within identified intercepted functions
-    for (size_t i = 0; i < to_map->curr_intercept_count; ++i)
+    for (unsigned short i = 0; i < to_map->curr_intercept_count; ++i)
```

Comment:
Is a `short` a good idea here?

Reply:
Since `curr_intercept_count` is itself short, I believe so.

I also noticed it wasn't unsigned, so I've now made it so 41fb04c639efcf1adac5d450557af8d1226654ef.